src/WebJobs.Script/Host/ProxyMetadataManager.cs

```diff
@@ -21,13 +21,13 @@ public class ProxyMetadataManager : IProxyMetadataManager
     {
         private static readonly Regex ProxyNameValidationRegex = new Regex(@"[^a-zA-Z0-9_-]", RegexOptions.Compiled | RegexOptions.IgnoreCase);
         private readonly Lazy<ProxyMetadataInfo> _metadata;
-        private readonly IOptions<ScriptJobHostOptions> _scriptOptions;
+        private readonly IOptions<ScriptApplicationHostOptions> _applicationHostOptions;
         private readonly IEnvironment _environment;
         private readonly ILogger _logger;
 
-        public ProxyMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IEnvironment environment, ILoggerFactory loggerFactory)
+        public ProxyMetadataManager(IOptions<ScriptApplicationHostOptions> applicationHostOptions, IEnvironment environment, ILoggerFactory loggerFactory)
         {
-            _scriptOptions = scriptOptions;
+            _applicationHostOptions = applicationHostOptions;
```

Comment:
Since this is moving to the root and will outlive the `JobHost` and specialization, we need to take an `IOptionsMonitor<ScriptApplicationHostOptions>` and update the usage below from `.Value` to `.CurrentValue`. We already have this all wired up for change detection/notification and you'll get the updated value if you're using that pattern (`ScriptApplicationHostOptions` will change after specialization)

Reply:
Also noticed this issue with WebFunctionsManager, but that's not related to your changes.

Reply:
I don't think I have any changes to make here? It was an IOptions before - I just changed it to be ScriptApplicationHostOptions which is web host level config that won't change.

Reply:
They will change, though, during specialization, those settings change and the options instance is invalidated (will trigger the creation of a new one if you ask for it). It's unlikely this would be an issue with this service, as we'll probably not have anything actually causing this to be created, but it's a risk.

For singleton services in the root container, if they're taking a dependency on `ScriptApplicationHostOptions`, they should use `IOptionsMonitor` instead, as that guarantees you won't have a stale, cached instance. This is not an issue for JobHost scoped services,that's why this was fine for this service before. The change here is simply to change the interface and access the options by using the `CurrentValue` property instead (which will give you the updated version if it ever changes)

Reply:
Ok, switched to use an options monitor